src/thermo/PureFluidPhase.cpp

```diff
@@ -90,7 +90,7 @@ std::string PureFluidPhase::phaseOfMatter() const
         return "supercritical";
     } else if (m_sub->TwoPhase() == 1) {
         return "liquid-gas-mix";
-    } else if (pressure() < m_sub->Ps()) {
+    } else if (m_sub->x() == 1) {
```

Comment:
Just curious how this change is any different than the existing condition. What was the problem with the existing code? I guess it was returning the wrong value in a few cases, but which cases and why does this fix it? I also think that the routine should return `liquid-gas-mix` if the state is exactly the saturated vapor.

Reply:
@bryanwweber ... if you set the state to saturated vapor, I.e. `Q = 1`, the old implementation returned `liquid`. I didn’t feel like opening an issue, and to just fix it.

Regarding your comment, this was my initial fix, but I then realized that several properties still make sense as being defined for saturated vapor or liquid (e.g. cp), but not the mix.

Reply:
> if you set the state to saturated vapor, I.e. `Q = 1`, the old implementation returned `liquid`

Yes, but why? :smile: Like I said, my preference would be to return `liquid-gas-mix` for the saturated vapor, I think that's the typical definition. The change here would rely on the quality being defined as equal to 1 for gases, which your other issue #916 would change...

Reply:
Correct - #916 would change it, in which case the correct check would be
``` C++
} else if (pressure() < m_sub->Ps() ||  m_sub->x() == 1) {
```

Regarding the definition, my first reaction was the same as yours, until the existing implementation reminded me of being able to define `cp`, etc. for the saturated lines based on one-sided differentials. Q=0 and 1 *are* liquid and vapor, after all, so it's not a clear cut case.

Reply:
**PS:** this is not a major sticking point for me. If we go with `liquid-gas-mix`, `cp` and a couple of other properties should likely return `nan` for the edge cases for consistency's sake?

Reply:
**PPS:** On further consideration, it would be odd to declare a "saturated *liquid*" with `Q=0` to be a `liquid-vapor-mix` instead of the `liquid` that defines the term; same goes for a "saturated *vapor*" ... so I *think* the current implementation makes the most sense?

Reply:
@bryanwweber ... I created #919 to facilitate the discussion.

Reply:
Reverting to the old implementation, while updating `TwoPhase` to include boundaries (a new flag `strict` excludes boundaries to enable property calculations).